### PR TITLE
Add family name to manage students table

### PR DESCRIPTION
**Pull request by rshipp**
Add the ability to view and add/edit/remove student family name in the section Manage Students table. Only shows in non-PL sections. **Behind a feature flag.**

image

## Links

- jira ticket: [TEACH-550](https://codedotorg.atlassian.net/browse/TEACH-550)

### CODE REVIEW

File: apps/src/templates/manageStudents/ManageStudentsTable.jsx [hunk 1]
```diff
@@ -484,6 +508,43 @@ class ManageStudentsTable extends Component {
...
+            children: (
+              <>
+                {React.cloneElement(sortable(label, columnInfo).children, {
+                  key: '1',
+                })}
+                <HelpTip key="2">
+                  <p>{i18n.familyNameHelpTip()}</p>
+                </HelpTip>
+              </>
+            ),
```
Comment (rshipp) [inline apps/src/templates/manageStudents/ManageStudentsTable.jsx:536]: this seems hacky, but it's the only way i could find to get the helptip into the header without losing the sorting buttons

**Comment by dju90** [comment]
Just double checking, this is behind a feature flag, correct?

**Comment by rshipp** [comment]
aah! I forgot that step! thank you! i'll add that in real quick

File: apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    editedValue: PropTypes.string,
+
+    //Provided by redux
+    editStudent: PropTypes.func.isRequired,
+  };
+
+  onChangeName = e => {
+    // Convert the empty string back to null before saving, so it doesn't
+    // add empty family names to students.
+    const newValue = e.target.value || null;
```
Comment (hannahbergam) [inline apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx:21]: Nice touch!

File: apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      <div>
+        {!this.props.isEditing && <div>{familyName}</div>}
+        {this.props.isEditing && (
+          <div>
+            <input
+              style={styles.inputBox}
+              // Since familyName is optional, explicitly prevent value from
+              // being set to undefined.
+              value={editedValue || ''}
+              onChange={this.onChangeName}
```
Comment (hannahbergam) [inline apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx:38]: Feels like a very beginner question, but I'm unfamiliar with when an onChange gets called for an input box. Is it once the cursor/focus leaves the box?

**Review by hannahbergam — APPROVED** [review]
Looks great- nice work! It would be a cherry on top (and not necessary for this PR) to have family name input in some sort of UI test/ an eyes snapshot to capture it but that feels like an end-of-project task:)

File: apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx (hunk 3 above)
Comment (rshipp) [reply apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx:38]: had to look this up, but yeah that's exactly right, it's when the focus leaves the element, if the value has changed: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event#text_input_element

**Comment by rshipp** [comment]
good call on adding an eyes test. i don't see an example in there of using a DCDO flag, but I assume that's something we can do? i would like to leave it for a followup since there's some research involved there

**Comment by hannahbergam** [comment]
> good call on adding an eyes test. i don't see an example in there of using a DCDO flag, but I assume that's something we can do? i would like to leave it for a followup since there's some research involved there

Makes sense to me! Also as a followup- could we eventually make this a functional component?

**Comment by rshipp** [comment]
i thought about that. If it's easy, it'd be nice to convert all of these cell/table components to functional. it does feel weird to be adding a new class component. :/ I'll make some followup tasks and see where we can prioritize them

**Pull request merged**